File: tests/fixtures/python-domain/src/payment/handler.py
"""HTTP handler for payment endpoints."""

from __future__ import annotations

from dataclasses import asdict
from typing import Any, Optional

from .entity import PaymentStatus
from .repository import PaymentFilter
from .service import PaymentService


class PaymentHandler:
    """Exposes HTTP endpoints for payments."""

    def __init__(self, service: PaymentService) -> None:
        self._service = service

    def list_payments(
        self,
        customer_id: Optional[str] = None,
        status: Optional[str] = None,
        page: int = 1,
        page_size: int = 20,
    ) -> dict[str, Any]:
        status_enum = PaymentStatus(status) if status else None
        filter_ = PaymentFilter(
            customer_id=customer_id,
            status=status_enum,
            page=page,
            page_size=page_size,
        )
        result = self._service.list_payments(filter_)
        return {
            "items": [
                {
                    "id": p.id,
                    "customer_id": p.customer_id,
                    "amount": p.amount.amount,
                    "currency": p.amount.currency,
                    "status": p.status.value,
                }
                for p in result.items
            ],
            "total_count": result.total_count,
            "page": filter_.page,
            "page_size": filter_.page_size,
        }

File: tests/fixtures/node-domain/src/payment/handler.ts
import { PaymentService } from "./service.js";
import { PaymentFilter } from "./repository.js";

interface Request {
  query: Record<string, string | undefined>;
}

interface Response {
  status(code: number): Response;
  json(body: unknown): void;
}

/** Handler exposes HTTP endpoints for payments. */
export class PaymentHandler {
  constructor(private readonly service: PaymentService) {}

  async listPayments(req: Request, res: Response): Promise<void> {
    const filter: PaymentFilter = {
      customerId: req.query["customer_id"],
      status: req.query["status"] as PaymentFilter["status"],
      page: parseInt(req.query["page"] ?? "1", 10),
      pageSize: parseInt(req.query["page_size"] ?? "20", 10),
    };

    try {
      const result = await this.service.listPayments(filter);
      res.status(200).json({
        items: result.items.map((p) => ({
          id: p.id,
          customer_id: p.customerId,
          amount: p.amount.amount,
          currency: p.amount.currency,
          status: p.status,
        })),
        total_count: result.totalCount,
        page: filter.page,
        page_size: filter.pageSize,
      });
    } catch {
      res
        .status(500)
        .json({ error: { code: "internal", message: "failed to list payments" } });
    }
  }
}
